bcdaworker/main.go

```diff
@@ -204,9 +204,24 @@ func writeBBDataToFile(bb client.APIClient, acoID string, acoCMSID string, cclfB
 	totalBeneIDs := float64(len(cclfBeneficiaryIDs))
 	failThreshold := getFailureThreshold()
 	failed := false
+	suppressedList := models.GetSuppressedBlueButtonIDs()
 
 	for _, cclfBeneficiaryID := range cclfBeneficiaryIDs {
 		blueButtonID, err := beneBBID(cclfBeneficiaryID, bb)
+		foundSuppressed := false
+
+		for _, suppressedBBID := range suppressedList {
+			if blueButtonID == suppressedBBID {
+				foundSuppressed = true
+				break
+			}
+		}
+
+		// skip over this cclf beneficiary if their blue button id is suppressed
+		if foundSuppressed {
+			continue
+		}
+
```

Comment:
If we're checking for suppression at this stage, does it still need to happen when the job is created? https://github.com/CMSgov/bcda-app/pull/439/files#diff-efd18a7d1940026f39dee80f50bb3bebR253 I'm wondering if the performance hit of redundant behavior is worth trying to keep the files equal(ish) in terms of beneficiary count.

Reply:
@embh 
Great comment.  We've been wrestling with this one... lots of discussions on this topic.  
One reason for the redundant check is that if this is an ACO's first pull after a new CCLF import, we won't have the BBIDs yet, so that first check won't actually remove suppressed benes (which is why we need the second check done by the workers).  If this is not the first pull after a new CCLF import, then we will have the BB ID, so the second check is indeed redundant.
We decided to keep the redundant because in March/April timeframe (we hope), we expect the suppression files to be compliant (i.e., use MBI instead of HICN).  When that occurs, we'll rip out the check done by the workers and revert back to the original code (but based on MBI not HICN).  
I agree there is a performance hit - not sure which is the more sound approach.  Definitely open to a lot more discussion.  Appreciate your thoughts.

Reply:
I forgot about the CCLF beneficiary records not having the BB IDs initially. That makes sense. Could a 1-800 file be ingested while export jobs are running? It might be preferable to leave the suppression logic later in the process so the most up-to-date sharing preferences are applied.

Reply:
> I forgot about the CCLF beneficiary records not having the BB IDs initially. That makes sense. Could a 1-800 file be ingested while export jobs are running? It might be preferable to leave the suppression logic later in the process so the most up-to-date sharing preferences are applied.

Yup, it is possible for the suppression data to be ingested while export jobs are running.  However, there is an XX hour period of time after a sharing preference is updated (I believe it is 72 hours), which we'd be well within if we considered the previous day's suppression data in the event new data came in while a job was running.